R/centrality.R

```diff
@@ -1072,12 +1072,12 @@ harmonic_centrality <- harmonic_centrality_cutoff_impl
 
 
 
-bonpow.dense <- function(graph, nodes = V(graph),
-                         loops = FALSE, exponent = 1,
+bonpow.dense <- function(graph, nodes = V(graph), loops = FALSE,
+                         exponent = 1, weights = NULL,
                          rescale = FALSE, tol = 1e-7) {
   ensure_igraph(graph)
 
-  d <- as_adj(graph)
+  d <- as_adj(graph, attr = weights)
```

Comment:
The `attr` parameter of `as_adj` behaves differently from typical `weights` parameter of other functions, so this is not correct. See #906. I'd like to have this fixed during the ongoing major upgrade to C/igraph 0.10.

I'm thinking that this PR could be much simpler if we postponed it until after the 0.10 upgrade is complete, but that might take quite a while.

If we won't want to do this, the solution would be a workaround like that used in `alpha_centrality` (i.e. duplicate the graph object and add the weights as an attribute—yes, this is an ugly hack).